### PR TITLE
Eigh numpy and torch frontends + tests

### PR DESCRIPTION
**Pull request by mobley-trent**
Partial solution. I was not able to load the tests in my codespace. I am requesting a code review
Issue number: https://github.com/unifyai/ivy/issues/10849

### CODE REVIEW

**Comment by AnnaTz** [comment]
I'm a little confused about the purpose of the PR. If you're trying to fix `linalg.test_tensorflow_eigh`, then that's not the way. The failures for torch, jax and numpy you see on the dashboard are related to the corresponding backends, not frontends. In other words, the tensorflow function `eigh` is passing for the tensorflow backend but failing for the torch, jax and numpy backends.

**Comment by mobley-trent** [comment]
The purpose of the PR is adding the frontends for `eigh` in torch and numpy and corresponding tests.

**Comment by AnnaTz** [comment]
Then it shouldn't be linked to this https://github.com/unifyai/ivy/issues/7982, but to the proper ToDo lists of pytorch and numpy.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py [hunk 1]
```diff
@@ -723,6 +723,51 @@ def test_torch_eig(
...
+# eigh
+@handle_frontend_test(
+    fn_tree="torch.linalg.eigh",
+    dtype_and_input=_get_dtype_and_square_matrix(),
+    test_with_out=st.just(False),
+)
+def test_torch_eigh(
+    *,
+    dtype_and_input,
+    UPLO=st.sampled_from(("L", "U")),
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:735]: This should be in `@handle_frontend_test` as you did for the numpy test.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py [hunk 2]
```diff
@@ -723,6 +723,51 @@ def test_torch_eig(
...
+    l, v = ret
+    front_l, front_v = frontend_ret
+
+    assert_all_close(
+        ret_np=v @ np.diag(l) @ np.linalg.inv(v),
+        ret_from_gt_np=front_v @ np.diag(front_l) @ np.linalg.inv(front_v),
+        rtol=1e-2,
+        atol=1e-2,
+        ground_truth_backend=frontend,
+    )
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:768]: Tests for `eigh` and `eig` (which I suppose you used as a reference) are different. Please have a look at `test_eigh` which is `ivy.eigh`'s test as a reference for how you should approach both numpy's and torch's frontend tests.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py (hunk 2 above)
Comment (mobley-trent) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:768]: I used `eigvalsh` as a reference since they are mostly the same function

Comment (mobley-trent) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:768]: How should I proceed ?

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by mobley-trent** [comment]
I have commits ready but I can't push

**Comment by AnnaTz** [comment]
Hi @mobley-trent, you need to follow the instructions here https://discord.com/channels/799879767196958751/983684937712558080/1080080255387189318

**Comment by mobley-trent** [comment]
I did `git push --force` but I didn't sync the `master` branch first. Is this fatal ?

**Comment by AnnaTz** [comment]
> I did `git push --force` but I didn't sync the `master` branch first. Is this fatal ? @AnnaTz

Hi there, you don't need to try and fix this. You should create a new a PR copying the code you've written in this one. Please see the link I attached above.
